gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/datafetcher/QueryServiceEntityFetcher.java

```diff
@@ -185,7 +197,12 @@ public EntityFetcherResponse getAggregatedMetrics(
 
     QueryRequest.Builder builder =
         constructSelectionQuery(requestContext, entitiesRequest, entityIdAttributes, aggregates);
-    adjustLimitAndOffset(builder, entitiesRequest.getLimit(), entitiesRequest.getOffset());
+    adjustLimitAndOffset(
```

Comment:
Not related to this change again but the getEntities and getAggregatedMetrics are very similar. Can we extract  common code out of these methods?

Reply:
Right. Will try to make it a part of separate PR, so that the regression impact is minimal and separated out